python.md

```diff
@@ -1332,3 +1352,35 @@ def test_some_longwinded_process(support_client, factory):
 ```
 
 You get the idea.
+
+### <a name="random-is-not-unique">Random is not unique</a>
+
+If you need to generate some unique values in a test, don't use random to create
+them because random != a sequence of unique numbers. Either use a counter or, if
+a counter won't do, save which random values you have already used and guard against
+using them more than once, for example:
+
+```python
+class UniqueInt:
+    def __init__(self):
+        self.used = set()
+
+    def get(self):
+        while v is None or v in self.used:
+            v = random.randint()
+
+        self.used.add(v)
+        return v
```

Comment:
I wonder if this is actually a solution to the race-condition problem. 

As I understand it, our test factories started using random data as factory-boy's [sequences](https://factoryboy.readthedocs.io/en/latest/introduction.html#sequences) are vulnerable to race-conditions when tests are run in parallel processes. I think the above solution is functionally equivalent to factory-boy's sequence functionality and so will suffer from the same problem as each process is unaware of the values used by other processes (but they are still coupled by the database).